src/app-logic/web-channel.js

```diff
@@ -80,10 +87,15 @@ type StatusQueryResponse = {|
   //    - GET_PROFILE
   //    - GET_SYMBOL_TABLE
   //    - QUERY_SYMBOLICATION_API
+  // Version 2:
+  //   Shipped in Firefox 121.
+  //   Adds support for the following message types:
+  //    - GET_EXTERNAL_POWER_TRACKS
   version?: number,
 |};
 type EnableMenuButtonResponse = void;
 type GetProfileResponse = ArrayBuffer | MixedObject;
+type GetExternalPowerTracksResponse = MixedObject;
```

Comment:
In the backend side it says that this returns an array. Should we also change thistype to an array of mixed objects?

Also is it possible to get null here?

Reply:
Anything that's not an array would be an error, but given the data comes from a fetch to an external server, I'm afraid "anything" is "possible".

`retrieveProfileFromBrowser` (in `receive-profile.js`) contains a big try/catch that will show a fatal error if something goes wrong. I'm not sure if that's enough or if we need some more specific error handling.

I was leaning on the side of "it's the user's fault if they put in about:config in the `devtools.performance.recording.power.external-url` pref something that returns bogus data, so it's fine to produce a fatal error", but I'm open to other opinions :).

Reply:
Oh yeah, I guess it's okay to fail if the external tool provides bogus data. That try/catch block should be okay.